modules/aws/src/smithy4s/aws/internals/AwsQueryRPCInterpreter.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.aws
+package internals
+
+import cats.MonadThrow
+import smithy4s.{Endpoint, Hints, Schema, ShapeId, StreamingSchema}
+import smithy4s.http.CodecAPI
+import smithy4s.kinds._
+
+// format: off
+/**
+ * An interpreter for unary operations in the AWS_QUERY protocol
+ */
+private[aws] class AwsQueryRPCInterpreter[Alg[_[_, _, _, _, _]], Op[_,_,_,_,_], F[_]](
+    service: smithy4s.Service.Aux[Alg, Op],
+    endpointPrefix: String,
+    awsEnv: AwsEnvironment[F],
+    contentType: String,
+    codecAPI: CodecAPI
+)(implicit F: MonadThrow[F])
+    extends PolyFunction5[Op, AwsCall[F, *, *, *, *, *]] {
+// format: on
+
+  def apply[I, E, O, SI, SO](
+      op: Op[I, E, O, SI, SO]
+  ): AwsCall[F, I, E, O, SI, SO] = {
+    val (input, endpoint) = service.endpoint(op)
+    awsEndpoints(amendEndpoint(endpoint)).toAwsCall(input)
+  }
+
+  private def amendEndpoint[I, E, O, SI, SO](
+      endpoint: Endpoint[Op, I, E, O, SI, SO]
+  ): Endpoint[Op, I, E, O, SI, SO] = {
+    new Endpoint[Op, I, E, O, SI, SO] {
+      def id: ShapeId = endpoint.id
+      def input: Schema[I] =
+        // TODO: Pass a service id and version
+        endpoint.input.addHints(smithy4s.internals.InputOutput.Input.widen)
```

Comment:
I'll have to pass a service id and version over here.
Will need to create a new hint very similar to `InputOutput`, except that `Input` part will be like a `case class Input(serviceId: String, version: String)`.

Reply:
I'm not 100% if I'm on the right track with the `Input(serviceId: String, version: String` implementation.

Reply:
Done